pkg/controller/component/metad_cluster.go

```diff
@@ -80,15 +83,12 @@ func (c *metadCluster) syncMetadWorkload(nc *v1alpha1.NebulaCluster) error {
 
 	oldWorkloadTemp, err := c.clientSet.Workload().GetWorkload(namespace, componentName, gvk)
 	if err != nil && !apierrors.IsNotFound(err) {
-		return fmt.Errorf("failed to get workload %s for cluster %s/%s, error: %s", componentName, namespace, ncName, err)
+		log.Error(err, "failed to get workload")
+		return err
 	}
 
 	notExist := apierrors.IsNotFound(err)
-
 	oldWorkload := oldWorkloadTemp.DeepCopy()
```

Comment:
If not found, the oldWorkloadTemp is not `nil` ?

Reply:
The rest of the process has been determined the nil case